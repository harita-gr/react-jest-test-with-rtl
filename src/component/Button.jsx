//Button.jsx

import React, { useState } from "react";

function Button({ label }) {
  const [count, setCount] = useState(0);

  const handleClick = () => {
    setCount((prev) => prev + 1);
  };

  return (
    <div>
      <button onClick={handleClick}>{label}</button>
      <p>Clicked {count} times</p>
    </div>
  );
}

export default Button;
