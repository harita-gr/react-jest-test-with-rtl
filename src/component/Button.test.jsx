//Button.test.jsx

import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("Button Component", () => {
  // Runs before each test
  beforeEach(() => {
    render(<Button label="Click Me" />);
  });

  test("renders button with correct label", () => {
    const button = screen.getByText("Click Me");
    expect(button).toBeInTheDocument();
  });

  test("increments counter on click", () => {
    const button = screen.getByText("Click Me");

    // Initial text
    expect(screen.getByText("Clicked 0 times")).toBeInTheDocument();

    // Simulate click
    fireEvent.click(button);
    expect(screen.getByText("Clicked 1 times")).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByText("Clicked 2 times")).toBeInTheDocument();
  });

  test("Button matches snapshot", () => {
    const { asFragment } = render(<Button label="Click Me" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
