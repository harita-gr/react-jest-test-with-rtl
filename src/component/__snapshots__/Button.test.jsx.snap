// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Click Me
    </button>
    <p>
      Clicked 0 times
    </p>
  </div>
</DocumentFragment>
`;
